gitlab/client.go

```diff
@@ -0,0 +1,55 @@
+/*
+Copyright 2020 The Knative Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package gitlab
+
+import (
+	"context"
+	"io/ioutil"
+	"net/http"
+	"path/filepath"
+
+	"github.com/xanzy/go-gitlab"
+)
+
+const (
+	AccessTokenKey = "accessToken"
+	VolumeName     = "gitlab-binding"
+	MountPath      = "/var/bindings/gitlab"
+)
+
+// ReadKey may be used to read keys from the secret bound by the GitLabBinding.
+func ReadKey(key string) (string, error) {
+	data, err := ioutil.ReadFile(filepath.Join(MountPath, key))
+	if err != nil {
+		return "", err
+	}
+	return string(data), nil
+}
+
+// AccessToken reads the file named accessToken that is mounted by the GitLabBinding.
+func AccessToken() (string, error) {
+	return ReadKey(AccessTokenKey)
+}
+
+// New instantiates a new gitlab client from the access token from the GitLabBinding
+func New(ctx context.Context) (*gitlab.Client, error) {
```

Comment:
This is awesome!  One realization I had this morning (and was chatting with @lberk about) was aligning the changes we make to the environment with existing tooling.  e.g. in the case of GitHubBinding, I think we should support the `hub` CLI.  Not sure if there is a Gitlab equivalent, or if it's auth is flexible enough to just trivially add this, but here's the Github issue: https://github.com/knative/eventing-contrib/issues/1185